stratosphere/pm/source/impl/pm_process_manager.cpp

```diff
@@ -0,0 +1,680 @@
+
+        inline bool ShouldDisableAslr(u32 launch_flags) {
+            const auto firmware_version = GetRuntimeFirmwareVersion();
+            return launch_flags & GET_FLAG_MASK(DisableAslr);
+        }
+
+#undef GET_FLAG_MASK
+
+        enum class ProcessEvent {
+            None           = 0,
+            Exited         = 1,
+            Started        = 2,
+            Exception      = 3,
+            DebugRunning   = 4,
+            DebugSuspended = 5,
+        };
+
+        enum class ProcessEventDeprecated {
+            None           = 0,
+            Exception      = 1,
+            Exited         = 2,
+            DebugRunning   = 3,
+            DebugSuspended = 4,
+            Started        = 5,
+        };
+
+        inline u32 GetProcessEventValue(ProcessEvent event) {
+            if (GetRuntimeFirmwareVersion() >= FirmwareVersion_500) {
+                return static_cast<u32>(event);
+            }
+            switch (event) {
+                case ProcessEvent::None:
+                    return static_cast<u32>(ProcessEventDeprecated::None);
+                case ProcessEvent::Exited:
+                    return static_cast<u32>(ProcessEventDeprecated::Exited);
+                case ProcessEvent::Started:
+                    return static_cast<u32>(ProcessEventDeprecated::Started);
+                case ProcessEvent::Exception:
+                    return static_cast<u32>(ProcessEventDeprecated::Exception);
+                case ProcessEvent::DebugRunning:
+                    return static_cast<u32>(ProcessEventDeprecated::DebugRunning);
+                case ProcessEvent::DebugSuspended:
+                    return static_cast<u32>(ProcessEventDeprecated::DebugSuspended);
+                default:
+                    std::abort();
+            }
+        }
+
+        /* Process Tracking globals. */
+        HosThread g_process_track_thread;
+        SessionManagerBase *g_process_waitable_manager = nullptr;
+
+        /* Process lists. */
+        ProcessList g_process_list;
+        ProcessList g_dead_process_list;
+
+        /* Global events. */
+        IEvent *g_process_event = nullptr;
+        IEvent *g_hook_to_create_process_event = nullptr;
+        IEvent *g_hook_to_create_application_process_event = nullptr;
+        IEvent *g_boot_finished_event = nullptr;
+
+        /* Process Launch synchronization globals. */
+        IEvent *g_process_launch_start_event = nullptr;
+        HosSignal g_process_launch_finish_signal;
+        Result g_process_launch_result = ResultSuccess;
+        LaunchProcessArgs g_process_launch_args = {};
+
+        /* Hook globals. */
+        std::atomic<ncm::TitleId> g_title_id_hook;
+        std::atomic<bool> g_application_hook;
+
+        /* Helpers. */
+        void ProcessTrackingMain(void *arg) {
+            /* This is the main loop of the process tracking thread. */
+
+            /* Create waitable manager. */
+            static auto s_process_waiter = WaitableManager(1);
```

Comment:
Why are you using auto here?
```
static WaitableManager s_process_waiter(1);
```

Reply:
WaitableManager isn't an instantiable class because templating. It's actually WaitableManager<DefaultOptions> or something. I use auto for that reason, since I prefer it to SessionManagerBase.